core/shell.py

```diff
@@ -528,14 +547,16 @@ def Main(lang, arg_r, environ, login_shell, loader, readline):
   # Pretend it came from somewhere else
   if flag.location_str is not None:
     src = source.Synthetic(flag.location_str)
+    assert line_reader is not None
     if flag.location_start_line != -1:
       line_reader.SetLineOffset(flag.location_start_line)
 
   arena.PushSource(src)
 
   # TODO: assert arena.NumSourcePaths() == 1
   # TODO: .rc file needs its own arena.
-  c_parser = parse_ctx.MakeOshParser(line_reader)
+  if line_reader is not None:
```

Comment:
please remove this condition / change to assert as well

i.e. I think this code should have the invariant that `line_reader` is valid, so I want to avoid extra conditionals

 (actually if we turned on strict None checking in  MyPy it would do that)

Reply:
ah nice catch. forgot about these other checks